ios/Classes/TiMapSnapshotter.m

```diff
@@ -0,0 +1,120 @@
+/**
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2009-2016 by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+
+#import "TiMapSnapshotter.h"
+#import "TiMapSnapshotterProxy.h"
+#import "TiMapViewProxy.h"
+#import "TiMapView.h"
+
+@implementation TiMapSnapshotter
+
+-(void)dealloc
+{
+    [super dealloc];
+    RELEASE_TO_NIL(options);
+    RELEASE_TO_NIL(snapshotter);
+}
+
+-(MKCoordinateRegion)regionFromDict:(NSDictionary*)dict
+{
+    CGFloat latitudeDelta = [TiUtils floatValue:@"latitudeDelta" properties:dict];
+    CGFloat longitudeDelta = [TiUtils floatValue:@"longitudeDelta" properties:dict];
+    CLLocationCoordinate2D center;
+    center.latitude = [TiUtils floatValue:@"latitude" properties:dict];
+    center.longitude = [TiUtils floatValue:@"longitude" properties:dict];
+    MKCoordinateRegion region_;
+    MKCoordinateSpan span;
+    span.longitudeDelta = longitudeDelta;
+    span.latitudeDelta = latitudeDelta;
+    region_.center = center;
+    region_.span = span;
+    return region_;
+}
+
+-(MKMapSnapshotOptions*)options
+{
+    if (!options) {
+        options = [[MKMapSnapshotOptions alloc] init];
+        [options setScale:[[UIScreen mainScreen] scale]];
+        CGRect bounds = [[UIScreen mainScreen] bounds];
+        [options setSize:bounds.size];
```

Comment:
We need flexible sizes as discussed. People want to use this API because of that feature, otherwise they could just use `toImage`.